core/src/main/java/org/geysermc/geyser/command/ExceptionHandlers.java

```diff
@@ -71,36 +73,45 @@ private void register() {
 
-        registerExceptionHandler(InvalidCommandSenderException.class, (src, e) -> {
+        registerExceptionHandler(InvalidCommandSenderException.class, (ctx, e) -> {
             // We currently don't use cloud sender type requirements anywhere.
             // This can be implemented better in the future if necessary.
             Type type = e.requiredSenderTypes().iterator().next(); // just grab the first
             String typeString = GenericTypeReflector.getTypeName(type);
-            src.sendLocaleString("geyser.command.invalid_sender", e.commandSender().getClass().getSimpleName(), typeString);
+            ctx.sender().sendLocaleString("geyser.command.invalid_sender", e.commandSender().getClass().getSimpleName(), typeString);
         });
 
         registerExceptionHandler(NoPermissionException.class, ExceptionHandlers::handleNoPermission);
 
         registerExceptionHandler(NoSuchCommandException.class,
-            (src, e) -> src.sendLocaleString("geyser.command.not_found"));
+            (ctx, e) -> ctx.sender().sendLocaleString("geyser.command.not_found"));
 
         registerExceptionHandler(ArgumentParseException.class,
-            (src, e) -> src.sendLocaleString("geyser.command.invalid_argument", e.getCause().getMessage()));
+            (ctx, e) -> ctx.sender().sendLocaleString("geyser.command.invalid_argument", e.getCause().getMessage()));
 
         registerExceptionHandler(CommandExecutionException.class,
-            (src, e) -> handleUnexpectedThrowable(src, e.getCause()));
+            (ctx, e) -> handleUnexpectedThrowable(ctx.sender(), e.getCause()));
 
         registerExceptionHandler(Throwable.class,
-            (src, e) -> handleUnexpectedThrowable(src, e.getCause()));
+            (ctx, e) -> handleUnexpectedThrowable(ctx.sender(), e.getCause()));
     }
 
-    private <E extends Throwable> void registerExceptionHandler(Class<E> type, BiConsumer<GeyserCommandSource, E> handler) {
-        controller.registerHandler(type, context -> handler.accept(context.context().sender(), context.exception()));
+    private <E extends Throwable> void registerExceptionHandler(Class<E> type, BiConsumer<CommandContext<GeyserCommandSource>, E> handler) {
+        controller.registerHandler(type, context -> handler.accept(context.context(), context.exception()));
     }
 
-    private static void handleNoPermission(GeyserCommandSource source, NoPermissionException exception) {
+    private static void handleNoPermission(CommandContext<GeyserCommandSource> context, NoPermissionException exception) {
+        // Just for Bedrock players that try to run commands they do not have the permissions for.
+        GeyserCommandSource source = context.sender();
+
+        // Let backend server receive & handle the command
+        if (CommandRegistry.STANDALONE_COMMAND_MANAGER && source instanceof GeyserSession session) {
+            session.sendCommand(context.rawInput().input());
```

Comment:
Do we also want to forward under NoSuchCommandException, since BedrockCommandRequestTranslator just checks for a matching root command? I haven't tested the behaviour.